lib/Cake/Cache/CacheEngine.php

```diff
@@ -181,7 +181,7 @@ public function key($key) {
 
 		$prefix = '';
 		if (!empty($this->_groupPrefix)) {
-			$prefix = vsprintf($this->_groupPrefix, $this->groups());
+			$prefix = md5(vsprintf($this->_groupPrefix, $this->groups()));
```

Comment:
I'm not sure the `vsprintf` is necessary now. This could be `md5(implode('_', $this->groups()))` and get the same effect.